metricflow-semantics/metricflow_semantics/query/validation_rules/metric_time_requirements.py

```diff
@@ -71,73 +59,44 @@ def __init__(  # noqa: D107
             resolve_group_by_item_result=resolve_group_by_item_result,
         )
 
-        self._metric_time_specs = tuple(
-            TimeDimensionSpec.generate_possible_specs_for_time_dimension(
-                time_dimension_reference=TimeDimensionReference(element_name=METRIC_TIME_ELEMENT_NAME),
-                entity_links=(),
-                custom_granularities=self._manifest_lookup.custom_granularities,
-            )
-        )
-        self._query_items_analysis_cache: LruCache[
-            Tuple[ResolverInputForQuery, MetricReference], QueryItemsAnalysis
-        ] = LruCache(128)
-
-    def _get_query_items_analysis(
-        self, query_resolver_input: ResolverInputForQuery, metric_reference: MetricReference
-    ) -> QueryItemsAnalysis:
-        cache_key = (query_resolver_input, metric_reference)
-        result = self._query_items_analysis_cache.get(cache_key)
-        if result is not None:
-            return result
-        result = self._uncached_query_items_analysis(query_resolver_input, metric_reference)
-        self._query_items_analysis_cache.set(cache_key, result)
-        return result
-
-    def _uncached_query_items_analysis(
-        self, query_resolver_input: ResolverInputForQuery, metric_reference: MetricReference
-    ) -> QueryItemsAnalysis:
-        has_agg_time_dimension = False
-        has_metric_time = False
-        scds: List[InstanceSpec] = []
-
-        valid_agg_time_dimension_specs = self._manifest_lookup.metric_lookup.get_valid_agg_time_dimensions_for_metric(
-            metric_reference
+        self._query_includes_metric_time = (
```

Comment:
lol this is so much better I didn't know you could do it like this. Next time I'll do it this way!